cadcdata/cadcdata/storageinv.py

```diff
@@ -0,0 +1,826 @@
+                             ' application will try to deduce it',
+                        required=False)
+    parser.add_argument('-e', '--encoding',
+                        help='MIME encoding to set in archive. If missing,'
+                             ' the application will try to deduce it',
+                        required=False)
+    parser.add_argument('-r', '--replace',
+                        help='replace existing files or fail if identifier '
+                             'is new')
+    parser.add_argument(
+        'identifier',
+        help='unique identifier (URI) given to the file in the CADC '
+             'Storage Inventory or a root identifier when multiple files'
+             'are uploaded at the same time')
+    parser.add_argument(
+        'src',
+        help='files or directories containing the files to be put. Multiple '
+             'sources require a root identifier (terminated in "/"). URIs '
+             'corresponding to each of the source file will be created as '
+             'root URI + filename.', nargs='+')
+    parser.epilog = (
+        'Examples:\n'
+        '- Use user certificate to replace a file specify the type\n'
+        '      cadcput --cert ~/.ssl/cadcproxy.pem -t "application/fits" \n'
+        '              -r cadc:TEST/myfile.fits myfile.fits\n'
+        '- Use default netrc file ($HOME/.netrc) to put two files files:\n'
+        '      cadcput -v -n cadc:TEST/ myfile1.fits.gz myfile2.fits.gz\n'
+        '- Use a different netrc file to put files from a directory to '
+        'directory dir:\n'
+        '      cadcput -d --netrc ~/mynetrc -s ivo:cadc.nrc.ca/cadc/minoc '
+        'cadc:TEST/ dir\n'
+        '- Connect as user to put files from multiple sources (prompt for\n'
+        '  password if user not in $HOME/.netrc):\n'
+        '      cadcput -v -u auser cadc:TEST/ myfile.fits.gz dir1 dir2')
+
+    args = parser.parse_args()
+    subject = net.Subject.from_cmd_line_args(args)
+
+    client = StorageInventoryClient(subject, args.service, host=args.host,
+                                    insecure=args.insecure)
+    files = []
+    for file in args.src:
+        if os.path.isdir(file):
+            # append all files in the directory
+            for i in os.listdir(file):
+                if not os.path.isdir(os.path.join(file, i)):
+                    files.append(os.path.join(file, i))
+        else:
+            files.append(file)
+    if (len(files) > 1) and not args.identifier.endswith('/'):
+        raise RuntimeError(
+            'A root identifier (ends in "/") is required to put multiple '
+            'files: {}'.format(args.identifier))
+
+    for file in files:
+        if len(files) > 1:
+            id = '{}/{}'.format(args.identifier.strip('/'),
+                                os.path.basename(file))
+        else:
+            id = args.identifier
+        execute_cmd(args, client.cadcput, {'id': id,
```

Comment:
Do you mean to limit the call to files of one type and one encoding - i.e. can only put gzipped fits files?  And then, if there are previews for those files, that has to be a separate call?

That's an ok approach, I'm just trying to clarify that it's being done on purpose.

Reply:
That's a limitation for sure but it's messy to do it in the command otherwise. One could rely on `magic` to provide that info for each file.
I'm not sure whether I should make the app accept type and encoding for one file only. That would seem unfair if the directory contains files of the same type. Maybe it should allow wildcards so that a user could specify files of the same type. Not sure what the best approach is here.

Reply:
I think this can be resolved with a bit of added help text for the command, stating something like "If `cadcputs` is invoked for multiple files, those files must have the same MIME type and MIME encoding."

Or something like that. That way, if a user pays any attention, they can't be surprised by the behaviour. IMHO.

Reply:
Help text updated.